tests/api/test_aws.py

```diff
@@ -303,39 +303,109 @@ def get_statements_by_sid(policy_document):
 )
 def test_grant_bucket_access(iam, users, resources):
     bucket_arn = 'arn:aws:s3:::test-bucket'
-    path_arns = [f'{bucket_arn}{resource}' for resource in resources]
+    path_arns_list = []
+    path_arns_object = []
+    if resources:
```

Comment:
I'm not quite following this test but is the `if` necessary? You're using list comprehension so no need to initialise these two variables with an empty list, e.g. 

have you tried this without the `if resources:`?

```
path_arns_list = [f'{bucket_arn}{resource}' for resource in resources]
path_arns_object = [f'{bucket_arn}{resource}/*' for resource in resources]
```

Reply:
I don't think this is necessary so I'll remove the `if`.